### PR TITLE
Fix PCI IDs

### PR DESCRIPTION
**Pull request by plbossart**
merge CometLake Kconfigs and add missing values for SOF and legacy drivers.

### CODE REVIEW

File: sound/soc/sof/intel/Kconfig [hunk 1]
```diff
@@ -206,50 +205,23 @@ config SND_SOC_SOF_ICELAKE
...
-
-config SND_SOC_SOF_COMETLAKE_S_SUPPORT
-	bool "SOF support for CometLake-S"
+config SND_SOC_SOF_COMETLAKE_LP_SUPPORT
+	bool "SOF support for CometLake"
+	select SND_SOC_SOF_COMETLAKE_SUPPORT
 	help
 	  This adds support for Sound Open Firmware for Intel(R) platforms
-	  using the Cometlake-H processors.
+	  using the Cometlake processors.
```
Comment (lyakh) [inline sound/soc/sof/intel/Kconfig:223]: Confused. Until now we had -S, -H, and -LP, right? These changes to Kconfig remove -H and -S, add a neutral one, and re-use the -LP. But above you still have `select SND_SOC_SOF_COMETLAKE_S if SND_SOC_SOF_COMETLAKE_S_SUPPORT` Typos? I guess the above should be `SND_SOC_SOF_COMETLAKE_SUPPORT`?

File: sound/soc/sof/intel/Kconfig [hunk 2]
```diff
@@ -26,7 +26,6 @@ config SND_SOC_SOF_INTEL_PCI
 	select SND_SOC_SOF_COFFEELAKE  if SND_SOC_SOF_COFFEELAKE_SUPPORT
 	select SND_SOC_SOF_ICELAKE     if SND_SOC_SOF_ICELAKE_SUPPORT
 	select SND_SOC_SOF_COMETLAKE   if SND_SOC_SOF_COMETLAKE_SUPPORT
-	select SND_SOC_SOF_COMETLAKE_S if SND_SOC_SOF_COMETLAKE_S_SUPPORT
```
Comment (lyakh) [inline sound/soc/sof/intel/Kconfig:30]: ok, patch splitting looks a bit off. You could just merge these (so far) 2 patches or, if you really want to split them, maybe first add a new generic CML Kconfig option, then convert all uses of old ones to new ones in the code, finally remove old Kconfig options. But I think just 1 patch would be ok and easy to understand for all of these.

File: sound/soc/sof/intel/Kconfig [hunk 3]
```diff
@@ -206,50 +205,23 @@ config SND_SOC_SOF_ICELAKE
...
-	select SND_SOC_SOF_HDA_COMMON
-	help
-	  This option is not user-selectable but automagically handled by
-	  'select' statements at a higher level
-
-config SND_SOC_SOF_COMETLAKE_S_SUPPORT
-	bool "SOF support for CometLake-S"
+config SND_SOC_SOF_COMETLAKE_LP_SUPPORT
+	bool "SOF support for CometLake"
+	select SND_SOC_SOF_COMETLAKE_SUPPORT
```
Comment (kv2019i) [inline sound/soc/sof/intel/Kconfig:220]: Should this be combined with above SOF_COMETLAKE_SUPPORT? If not, then description should state CML-LP.
UPDATE: hmm, I guess this is for backwards compatibility (and S/H are not in upstream so thus the special treatment for LP). Some mention in comments would be good to mark old options that are just for backwards compatibility.

File: sound/soc/sof/intel/Kconfig (hunk 2 above)
Comment (kv2019i) [reply sound/soc/sof/intel/Kconfig:30]: @plbossart Yup, maybe make a series that reverts the non-upstream CML-S patch, then 2nd patch to unify the ifdefs for CML and final patch that simply adds the PCI ID for CML-S. Then we could just send patches 2 and 3 to upstream.

Comment (plbossart) [reply sound/soc/sof/intel/Kconfig:30]: ok, will add a revert for clarity

File: sound/soc/sof/intel/Kconfig (hunk 3 above)
Comment (kv2019i) [reply sound/soc/sof/intel/Kconfig:220]: @plbossart Should we have a mention in git commit that we have backwards compatibility for LP as that is widely used, but we don't have it for CML-H (that option is just dropped here). Is the rationale that CML-H typically does not have DSP enabled, or that existing users have had both CML options enabled and keeping LP around is sufficient?

Comment (plbossart) [reply sound/soc/sof/intel/Kconfig:220]: it's the latter, reworded the commit message to clarify this point.

**Comment by plbossart** [comment]
> In fact I'm not sure how valuable backward compatibility of Kconfig files is. At least I see regularly Kconfig files change, presumably causing hopefully minor annoyances to distros, so far I personally didn't consider this a part of a "stable API"

the ask is that makeoldconfig provides the same capabilities when updating, I took flak in the past for breaking configurations.

**Pull request merged**